platform/iphone/gl_view.mm

```diff
@@ -379,369 +222,106 @@ - (void)destroyFramebuffer {
+		} else {
+			// Approximate frame rate: assumes device refreshes at 60 fps.
+			// Note that newer iOS devices are 120Hz screens
+			_displayLink.frameInterval = 1;
+		}
 
 		// Setup DisplayLink in main thread
-		[displayLink addToRunLoop:[NSRunLoop currentRunLoop] forMode:NSRunLoopCommonModes];
+		[_displayLink addToRunLoop:[NSRunLoop currentRunLoop]
+						   forMode:NSRunLoopCommonModes];
 	} else {
-		animationTimer = [NSTimer scheduledTimerWithTimeInterval:animationInterval target:self selector:@selector(drawView) userInfo:nil repeats:YES];
+		// This is an extremely terrible way to animate. Very low resolution with lots
+		// of drift. You would be lucky to have this called every 16ms
+		animationTimer = [NSTimer scheduledTimerWithTimeInterval:self.animationInterval
+														  target:self
+														selector:@selector(drawView)
+														userInfo:nil
+														 repeats:YES];
 	}
 
-	if (video_playing) {
-		_unpause_video();
-	}
+	// if (video_playing) {
+	// 	_unpause_video();
+	// }
 }
 
 - (void)stopAnimation {
-	if (!active)
-		return;
-	active = FALSE;
-	printf("******** stop animation!\n");
+	if (!self.isActive) return;
+
+	self.active = NO;
+	printf("stop animation!\n");
 
-	if (useCADisplayLink) {
-		[displayLink invalidate];
-		displayLink = nil;
+	if (self.useCADisplayLink) {
+		[_displayLink invalidate];
+		_displayLink = nil;
 	} else {
 		[animationTimer invalidate];
 		animationTimer = nil;
 	}
 
-	clear_touches();
-
-	if (video_playing) {
-		// save position
-	}
+	// if (video_playing) {
+	// 	_pause_video();
+	// }
 }
 
 - (void)setAnimationInterval:(NSTimeInterval)interval {
-	animationInterval = interval;
-	if ((useCADisplayLink && displayLink) || (!useCADisplayLink && animationTimer)) {
+
+	_animationInterval = interval;
+
+	if (self.isActive) {
 		[self stopAnimation];
-		[self startAnimation];
 	}
+	[self startAnimation];
 }
 
-// Updates the OpenGL view when the timer fires
-- (void)drawView {
-	if (useCADisplayLink) {
-		// Pause the CADisplayLink to avoid recursion
-		[displayLink setPaused:YES];
-
-		// Process all input events
-		while (CFRunLoopRunInMode(kCFRunLoopDefaultMode, 0, TRUE) == kCFRunLoopRunHandledSource)
-			;
-
-		// We are good to go, resume the CADisplayLink
-		[displayLink setPaused:NO];
-	}
+- (NSTimeInterval)animationInterval {
+	return _animationInterval;
+}
 
-	if (!active) {
-		printf("draw view not active!\n");
+- (void)drawView {
+	if (!self.isActive) {
+		printf("Attempted to drawView while inactive!\n");
 		return;
-	};
+	}
 
 	// Make sure that you are drawing to the current context
 	[EAGLContext setCurrentContext:context];
 
 	// If our drawing delegate needs to have the view setup, then call -setupView: and flag that it won't need to be called again.
-	if (!delegateSetup) {
-		[delegate setupView:self];
-		delegateSetup = YES;
+	if (!self.isSetUpComplete && [self.delegate respondsToSelector:@selector(setupView:)]) {
```

Comment:
This should only be called once on start up. It's an optional delegate method to allow customization of the OpenGL environment.

Reply:
It's unimplemented here.